components/logins/ffi/src/lib.rs

```diff
@@ -334,49 +353,57 @@ pub extern "C" fn sync15_passwords_get_by_id(
     })
 }
 
+/// # Safety
+/// Deref pointer, thus unsafe
 #[no_mangle]
-pub extern "C" fn sync15_passwords_add(
+pub unsafe extern "C" fn sync15_passwords_add(
     handle: u64,
-    record_json: FfiStr<'_>,
+    data: *const u8,
+    len: i32,
     error: &mut ExternError,
 ) -> *mut c_char {
     log::debug!("sync15_passwords_add");
     ENGINES.call_with_result(error, handle, |state| {
-        let mut parsed: serde_json::Value = serde_json::from_str(record_json.as_str())?;
-        if parsed.get("id").is_none() {
```

Comment:
Note for myself: in practice this condition was always `false` as `id` is always serialized.